src/conf.c

```diff
@@ -135,8 +135,10 @@ static int validate_common(cfg_t *cfg, const char *provider, int custom)
 	}
 
 	if (!custom && !cfg_getstr(cfg, "username")) {
-		cfg_error(cfg, "Missing username setting for DDNS provider %s", provider);
-		return -1;
+		if (strcmp(provider, "default@freemyip.com") != 0) {	
```

Comment:
Hmm, I see, didn't realize that at first glance.

OK, that'll need something entirely different to support ... not entirely happy with the strcmp() and a hard-coded provider.  But OK, let's merge this and I'll sprinkle some fairy dust on top later to make it go away ;-)